### ISSUE TITLE
Case insensitive use of association names in AssociationCollection causes problems.

### ISSUE THREAD

**Issue by ADmad** [cakephp/cakephp#14554]
I have seen this issue come up many times in our support channels. 

### The problem
This diff below shows how the result gets changed if one mistakenly uses incorrect inflection when containing an association. If one uses `authors` instead of `Authors` in `contain()` the related record is now under property `Authors` instead of `author` and also it's an array instead of entity.

```php
diff --git a/tests/TestCase/ORM/Association/BelongsToTest.php b/tests/TestCase/ORM/Association/BelongsToTest.php
index 3fe85b2229..6cc56e870f 100644
--- a/tests/TestCase/ORM/Association/BelongsToTest.php
+++ b/tests/TestCase/ORM/Association/BelongsToTest.php
@@ -415,12 +415,12 @@ class BelongsToTest extends TestCase
         $query = $articles->find()
             ->select(['Authors.name'])
             ->where(['Articles.id' => 1])
-            ->contain('Authors');
+            ->contain('authors');
         $result = $query->firstOrFail();
 
-        $this->assertNotEmpty($result->author);
-        $this->assertSame('mariano', $result->author->name);
-        $this->assertSame(['author'], array_keys($result->toArray()), 'No other properties included.');
+        $this->assertNotEmpty($result->Authors);
+        $this->assertSame('mariano', $result->Authors['name']);
+        $this->assertSame(['Authors'], array_keys($result->toArray()), 'No other properties included.');
     }
 
     /**
```

### The cause
The source of the problem is that all methods of `AssociationCollection` lower case the association names.

### The solution
Remove the lower casing of assocation names/aliases. That would cause the ORM to throw exception with message like `"The authors association is not defined on Articles."` so users are made aware of the problem and they can fix it.

**Comment by dereuromark** [comment]
Shouldnt this go at least into 4.next milestone?
It can be quite bc breaking for people.

**Comment by ADmad** [comment]
Yes the fix would have to go into 4.1

**Comment by othercorey** [comment]
Fixed in https://github.com/cakephp/cakephp/pull/14556
